components/script/dom/readablestream.rs

```diff
@@ -482,6 +489,35 @@ impl ReadableStream {
 
         // TODO: react to sourceCancelPromise.
     }
+
+    /// <https://streams.spec.whatwg.org/#readable-stream-reader-generic-release>
+    pub fn release_lock(&self, closed_promise: &Promise) {
+        // step 1 & 2
+
+        if self.is_readable() {
+            // step 3
+            closed_promise.reject_error(Error::Type("stream state is not readable".to_owned()));
+        } else {
+            // step 4
+            closed_promise.reject_error(Error::Type(
```

Comment:
indeed I missed this:

Otherwise, set reader.[[[closedPromise]]](https://streams.spec.whatwg.org/#readablestreamgenericreader-closedpromise) to [a promise rejected with](https://webidl.spec.whatwg.org/#a-promise-rejected-with) a [TypeError](https://tc39.es/ecma262/#sec-native-error-types-used-in-this-standard-typeerror) exception.

thanks